code/WorkInProgress/Cael_Aislinn/Rust/fuel_control.dm

```diff
@@ -121,11 +107,12 @@
 /obj/machinery/computer/rust_fuel_control/Topic(href, href_list)
 	if(..()) return 1
 
-	if( href_list["scan"] )
-		connected_injectors = list()
-		for(var/obj/machinery/power/rust_fuel_injector/I in range(scan_range, src))
-			if(check_injector_status(I))
-				connected_injectors.Add(I)
+	if( href_list["toggle_injecting"])
```

Comment:
The way this is currently set up toggles **all** injectors. 

Do the following (assuming you listened to my previous comment):

``` DM
var/obj/machinery/power/rust_fuel_injector/I = locate(href_list["toggle_injecting"]) 
if(!istype(I))
	return 1
```

Then toggle `I`